content/en/security/application_security/enabling/compatibility/java.md

```diff
@@ -6,45 +6,139 @@ type: multi-code-lang
+| ---           |   ---             |           ----            |
+| Docker        | {{< X >}}         | {{< X >}}                 |
+| Kubernetes    | {{< X >}}         | {{< X >}}                 | 
+| AWS ECS       | {{< X >}}         | {{< X >}}                 |
+| AWS Fargate   | {{< X >}}         | {{< X >}}                 |
+| AWS Lambda    | {{< X >}}         |                           |   
+
+
 ## Language and framework compatibility
 
 ### Supported Java versions
+The Java Tracer supports automatic instrumentation for the following Oracle JDK and OpenJDK JVM runtimes.
+
+| JVM versions | Operating Systems                                                               | Support level                       | Tracer version |
+| -------------| ------------------------------------------------------------------------------- | ----------------------------------- | -------------- |
+| 8 to 17      | Windows (x86-64)<br>Linux (glibc, musl) (arm64, x86-64)<br>MacOS (arm64, x86-64)               | Supported                | Latest         |
 
-The Datadog library supports Java JRE 1.8 and higher of both Oracle JDK and OpenJDK, on the following architectures:
-- Linux (GNU) x86, x86-64
-- Alpine Linux (musl) x86, x86-64
-- macOS (Darwin) x86, x86-64
-- Windows (msvc) x86, x86-64
 
 Datadog does not officially support any early-access versions of Java.
 
-You can monitor application security for Java apps running in Docker, Kubernetes, AWS ECS, and AWS Fargate.
 
-### Supported frameworks
 
-| Framework Web Server    | Minimum Framework Version   |
-| ----------------------- | --------------------------- |
-| Servlet Compatible      | 2.3+, 3.0+                  |
-| Spring                  | 3.1                         |
 
-**Note**: Many application servers are Servlet compatible and are supported by ASM, such as WebSphere, WebLogic, and JBoss. Also, frameworks like Spring Boot are supported by virtue of using a supported embedded application server (such as Tomcat, Jetty, or Netty).
+                                                                                                                                 
+
+### Web framework compatibility
+
+- Attacker source HTTP request details
+- Tags for the HTTP request (status code, method, etc)
+- Distributed Tracing to see attack flows through your applications
+
+##### ASM Capability Notes
+- Open Source Vulnerability Detections is supported on all frameworks
+
+
+
+| Framework                  | Versions   | Threat Detection supported? | Threat Protection supported? | Custom Code Vulnerability Detection supported? |
+| ----------------------- | ---------- | --------------- | ---------------------------------------------- | ---------------------------------------------- |
+| Grizzly                 | 2.0+       |  {{< X >}} |  {{< X >}} |  {{< X >}} |
+| Glassfish               |            |  {{< X >}} |  {{< X >}} |  {{< X >}} |
```

Comment:
Supported for Threat Detection in 4.x and 5.x (but not higher).  Supported for IAST (tested with latest Payara, but the support should pretty much rely on general servlet compatibility).

Reply:
thanks!